text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,338 @@
+---
+title: Dynamic Component Creation
+status: DRAFTED
+created_at: 2022-12-21
+updated_at: 2023-01-04
+pr: (leave this empty until the PR is created)
+---
+
+# Dynamic Component Creation
+
+## Summary
+
+This RFC supersedes the existing `lwc:dynamic` directive to introduce a new way of creating dynamic components without reusing the custom element name.
+
+Dynamic components refer to custom elements where the constructor is not known at compile time.
+
+*This proposal only focuses on the dynamic component creation process, lazy loading which was included in the [original proposal for dynamic components](https://rfcs.lwc.dev/rfcs/lwc/0110-dynamic-components) is not included.*
+
+## Basic example
+
+```html
+<template>
+    <div class="container">
+        <lwc:dynamic lwc:is={lazyConstructor}></lwc:dynamic>
+    </div>
+</template>
+```
+
+```javascript
+import { LightningElement } from 'lwc';
+
+export default class extends LightningElement {
+    lazyConstructor;
+
+    async connectedCallback() {
+        const { default: ctor } = await import('lightning/concreteComponent');
+        this.lazyConstructor = ctor;
```

Comment:
This dynamic import and `async` / `await` could confuse the reader (one of my coworkers in France wasn't clear about this). I would recommend either stripping this from this example or adding another example where the constructor is imported sync just to nail the point.

Reply:
agreed! also, if their assign a promise to the `lazyConstructor` by mistake, it should throw, It must be a qualifying constructor or undefined, make sure you mentioned that.

Reply:
All examples have been updated and there is a note about using promises [here](https://github.com/salesforce/lwc-rfcs/pull/71/files#diff-a1de05148a61ea9b00f70bd4ff5feb3895c16dd99707b173ba57c081b859250bR218).